### ISSUE TITLE
dep-graph seems to be caching result and not updating

### ISSUE THREAD

**Issue by markgoho** [nrwl/nx#2445]
## Expected Behavior

`npm run dep-graph` shows the dependency graph

## Current Behavior

connections seem to be accurate, but when removing an `implicitDependency` from the `nx.json` file, the connection is still rendering as if it existed

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. run `npm dep-graph`
2. Add an implicitDependency from one lib to another
3. run `npm dep-graph` again, see the new connection
4. remove that dependency
5. run `npm dep-graph` again and see the connection still rendered

### Context

Please provide any relevant information about your setup:

``` bash
@nrwl/angular : 8.12.2
@nrwl/cli : 8.12.2
@nrwl/cypress : 8.12.2
@nrwl/eslint-plugin-nx : Not Found
@nrwl/express : Not Found
@nrwl/jest : 8.12.2
@nrwl/linter : Not Found
@nrwl/nest : Not Found
@nrwl/next : Not Found
@nrwl/node : Not Found
@nrwl/react : Not Found
@nrwl/schematics : Not Found
@nrwl/tao : 8.12.2
@nrwl/web : Not Found
@nrwl/workspace : 8.12.2
typescript : 3.5.3
```

## Other

From the nx.json:
``` json
"projects": {
  "theme": {
      "tags": ["type:theme", "scope:shared"]
    },
    "admin": {
      "tags": ["type:app", "scope:admin"]
    },
}
```

and the `window.graph` object in the dep-graph view:
``` json
"dependencies": {
  "admin": [
      { "type": "static", "source": "admin", "target": "admin-shell" },
      { "type": "implicit", "source": "admin", "target": "theme" }
    ],
    "theme": [
      { "type": "implicit", "source": "theme", "target": "admin-shell" }
    ],
}
```

Clearly you can see that the dep-graph object has been created incorrectly, based on a previous value that no longer exists in the `nx.json` file.

Video showing the behavior:
https://www.youtube.com/watch?v=-Q68s1zCdN4&feature=youtu.be

**Comment by markgoho** [comment]
Here's a screen recording of the issue: https://youtu.be/bc0LvOojTU4

**Comment by x87** [comment]
The cached version of the project graph is stored into `dist/nxdeps.json`

https://github.com/nrwl/nx/blob/b04b18d07ef77bd2141472714afa24c799ea1dff/packages/workspace/src/core/project-graph/project-graph.ts#L90

**Comment by markgoho** [comment]
Great catch @x87 -- I saw this as well in my project. Could this be a simple fix of deleting that file each time the command is run?

**Comment by vsavkin** [comment]
This has been fixed in Nx 8.12.7 and 9.0.2.
